protocol/x/clob/keeper/clob_pair.go

```diff
@@ -198,6 +188,35 @@ func (k Keeper) InitMemClobOrderbooks(ctx sdk.Context) {
 	}
 }
 
+// HydrateClobPairAndPerpetualMapping hydrates the in-memory mapping between clob pair and perpetual.
+func (k Keeper) HydrateClobPairAndPerpetualMapping(ctx sdk.Context) {
+	clobPairs := k.GetAllClobPairs(ctx)
+	for _, clobPair := range clobPairs {
+		// Create the corresponding mapping between clob pair and perpetual.
+		k.SetClobPairIdForPerpetual(
+			ctx,
+			clobPair,
+		)
+	}
+}
+
+// SetClobPairIdForPerpetual sets the mapping between clob pair and perpetual.
+func (k Keeper) SetClobPairIdForPerpetual(ctx sdk.Context, clobPair types.ClobPair) {
+	// If this `ClobPair` is for a perpetual, add the `clobPairId` to the list of CLOB pair IDs
+	// that facilitate trading of this perpetual.
+	if perpetualClobMetadata := clobPair.GetPerpetualClobMetadata(); perpetualClobMetadata != nil {
+		perpetualId := perpetualClobMetadata.PerpetualId
+		clobPairIds, exists := k.PerpetualIdToClobPairId[perpetualId]
+		if !exists {
+			clobPairIds = make([]types.ClobPairId, 0)
+		}
```

Comment:
Should we panic if it already exists, since that indicates there is more than one CLOB pair for a perp?

Reply:
we panic when we call the getter function  `GetClobPairForPerpetual` so I think we are fine